File: next/kmp/core/src/commonMain/kotlin/org/dweb_browser/core/ipc/helper/IpcLifeCycle.kt
package org.dweb_browser.core.ipc.helper

import kotlinx.serialization.Serializable

/**
 * Ipc生命周期控制
 */
@Serializable
class IpcLifeCycle(
  val state: IPC_STATE,
  val encoding: List<IPC_DATA_ENCODING> = listOf()
) : IpcMessage(IPC_MESSAGE_TYPE.LIFE_CYCLE) {
  override fun toString() = "IpcLifeCycle(state=$state)"

  companion object {
    fun opening() = IpcLifeCycle(IPC_STATE.OPENING)
    fun open() = IpcLifeCycle(IPC_STATE.OPEN)

    fun closing() = IpcLifeCycle(IPC_STATE.OPENING)
    fun close() = IpcLifeCycle(IPC_STATE.CLOSED)
  }

}

File: desktop-dev/src/core/ipc/IpcLifeCycle.ts
import { IPC_DATA_ENCODING, IPC_MESSAGE_TYPE, IPC_STATE } from "./helper/const.ts";
import { IpcMessage } from "./helper/IpcMessage.ts";

export class IpcLifeCycle extends IpcMessage<IPC_MESSAGE_TYPE.LIFE_CYCLE> {
  constructor(readonly state: IPC_STATE, readonly encoding: IPC_DATA_ENCODING[] = []) {
    super(IPC_MESSAGE_TYPE.LIFE_CYCLE);
  }

  static opening = () => new IpcLifeCycle(IPC_STATE.OPENING);

  static open = () => new IpcLifeCycle(IPC_STATE.OPEN);

  static closing = () => new IpcLifeCycle(IPC_STATE.OPENING);
  static close = () => new IpcLifeCycle(IPC_STATE.CLOSED);
}
